### PR TITLE
[openxr-loader] update to 1.0.31

### PR DESCRIPTION
**Pull request by jherico**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
  - I think?  
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/openxr-loader/portfile.cmake [hunk 1]
```diff
@@ -2,8 +2,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO KhronosGroup/OpenXR-SDK
-    REF release-1.0.22
-    SHA512 fe3c393c2d11981b42355acd8dbc337727120bcd0ff595abac1975c4ce5f68bb74a9a1b4c959e64e9a847ae5d504100d31979ffd7d9702c55b2dbd889de17d3e
+    REF release-1.0.31
```
Comment (Cheney-W) [inline ports/openxr-loader/portfile.cmake:5]:
```suggestion
    REF "release-${VERSION}"
```

File: ports/openxr-loader/portfile.cmake [hunk 2]
```diff
@@ -12,8 +12,8 @@ vcpkg_from_github(
 vcpkg_from_github(
     OUT_SOURCE_PATH SDK_SOURCE_PATH
     REPO KhronosGroup/OpenXR-SDK-Source
-    REF release-1.0.22
-    SHA512 92802d57a45ca1d697d3cea1b3f5619af4ba36156cb28c2c39b2295a74ebc45907caf371c916c54ec3be44a2f3ae447ffc1cd62f54b7b24f7a081408328c7651
+    REF release-1.0.31
```
Comment (Cheney-W) [inline ports/openxr-loader/portfile.cmake:15]:
```suggestion
    REF "release-${VERSION}"
```

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -900,6 +900,9 @@ openturns:x64-osx=fail
 openvpn3:x64-osx=fail
 openvpn3:arm64-osx=fail
 openvr:x64-windows-static=fail
+openxr-loader:arm-neon-android=fail
+openxr-loader:x64-android=fail
+openxr-loader:arm64-android=fail
```
Comment (Cheney-W) [inline scripts/ci.baseline.txt:905]: Why add these records instead of using `"supports": "!uwp & !osx & !android"` in `vcpkg.json` file?

**Comment by Cheney-W** [comment]
When you respond, please revert to "ready for review".

File: scripts/ci.baseline.txt (hunk 3 above)
Comment (jherico) [reply scripts/ci.baseline.txt]: Locally executing `./vcpkg.exe install --triplet arm64-android openxr-loader` works fine for me, using NDK 21.4.7075529.  I will try to dig in more as to why the CI builds are failing on Android.

**Comment by jherico** [comment]
At some point either the `hpp_gen.py` script I'm using the the portfile changed to write it's output to stderr OR the CI process started treating any stderr output as a failure.  Hopefully this should fix it.... and as I'm typing it the android build has already failed.

**Comment by jherico** [comment]
OK, I finally found the issue.  It looks like the Android CI docker images are all running a version of Python < 3.9 (they're still based on Ubuntu 20.04).  The script used to generate the C++ bindings contains a type annotation that's incompatible with Python 3.8, so I've added a patch to remove the annotation (I don't really want to push this back upstream to the source repository, because honestly Python 3.8 is a bit old).

Should be good to go.

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**